mkosi/distributions/ubuntu.py

```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+from typing import Set
+
+from ..backend import MkosiState, add_packages
+from .debian import DebianInstaller
+
+
+class UbuntuInstaller(DebianInstaller):
+    repositories_for_boot = {"universe"}
+
+    @classmethod
+    def add_default_kernel_package(cls, state: MkosiState, extra_packages: Set[str]) -> None:
+        # use the global metapckage linux-generic if the user didn't pick one
+        if ("linux-generic" not in extra_packages and
+            not any(package.startswith("linux-image") for package in extra_packages)):
+            add_packages(state.config, extra_packages, "linux-generic")
+
+    @classmethod
+    def add_apt_auxiliary_repos(cls, state: MkosiState, repos: Set[str]) -> None:
```

Comment:
We already have the `Repo` concept for RPM distros, we can easily do the same here and pass those repo definitions to a shared install function.

Reply:
In fact, I don't see why we can't do this outside of the shared install function. If no local mirror is specified, write the auxiliary repos before calling the shared install function. If one is specified, write the shared repos after calling the shared install function.